helpers/templates/time_test.go

```diff
@@ -119,23 +73,60 @@ func TestHumanizeDurationSubsecondAndFractionalSecondsString(t *testing.T) {
 	}
 }
 
-func TestHumanizeDurationSecondsInt(t *testing.T) {
+func TestHumanizeDurationErrorString(t *testing.T) {
+	_, err := HumanizeDuration("one")
+	require.Error(t, err)
+}
+
+func TestHumanizeTimestamp(t *testing.T) {
 	tc := []struct {
 		name     string
-		input    int
+		input    interface{}
 		expected string
 	}{
-		{name: "zero", input: 0, expected: "0s"},
-		{name: "negative", input: -1, expected: "-1s"},
-		{name: "second", input: 1, expected: "1s"},
-		{name: "days", input: 1234567, expected: "14d 6h 56m 7s"},
+		// Int
+		{name: "zero", input: 0, expected: "1970-01-01 00:00:00 +0000 UTC"},
+		{name: "negative", input: -1, expected: "1969-12-31 23:59:59 +0000 UTC"},
+		{name: "one", input: 1, expected: "1970-01-01 00:00:01 +0000 UTC"},
+		{name: "past", input: 1234567, expected: "1970-01-15 06:56:07 +0000 UTC"},
+		{name: "future", input: 9223372036, expected: "2262-04-11 23:47:16 +0000 UTC"},
+		// Uint
+		{name: "zero", input: uint64(0), expected: "1970-01-01 00:00:00 +0000 UTC"},
+		{name: "one", input: uint64(1), expected: "1970-01-01 00:00:01 +0000 UTC"},
+		{name: "past", input: uint64(1234567), expected: "1970-01-15 06:56:07 +0000 UTC"},
+		{name: "future", input: uint64(9223372036), expected: "2262-04-11 23:47:16 +0000 UTC"},
+		// NaN/Inf, strings
+		{name: "infinity", input: "+Inf", expected: "+Inf"},
+		{name: "minus infinity", input: "-Inf", expected: "-Inf"},
+		{name: "NaN", input: "NaN", expected: "NaN"},
+		// Nan/Inf, float64
+		{name: "infinity", input: math.Inf(1), expected: "+Inf"},
+		{name: "minus infinity", input: math.Inf(-1), expected: "-Inf"},
+		{name: "NaN", input: math.NaN(), expected: "NaN"},
 	}
 
 	for _, tt := range tc {
 		t.Run(tt.name, func(t *testing.T) {
-			result, err := HumanizeDuration(tt.input)
+			result, err := HumanizeTimestamp(tt.input)
 			require.NoError(t, err)
 			require.Equal(t, tt.expected, result)
 		})
 	}
 }
+
+func TestHumanizeTimestampError(t *testing.T) {
+	_, err := HumanizeTimestamp(math.MaxInt64)
+	require.Error(t, err)
+}
+
+func TestHumanizeTimestampSampleFloat64(t *testing.T) {
+	result, err := HumanizeTimestamp(1435065584.128)
+	require.NoError(t, err)
+	require.Equal(t, "2015-06-23 13:19:44.128 +0000 UTC", result)
+}
+
+func TestHumanizeTimestampSampleString(t *testing.T) {
+	result, err := HumanizeTimestamp(1435065584.128)
+	require.NoError(t, err)
+	require.Equal(t, "2015-06-23 13:19:44.128 +0000 UTC", result)
+}
```

Comment:
This could also be a single test, but that's just nitpicking :P

Reply:
Actually yeah, can fix right away

Reply:
Also done, should be better now. Thanks for suggesting!
Also found a bug in one of these tests (float64 instead of string), fixed it there as well